apps/client/src/features/bookmark/bookmar-modal-import/ui/bookmark-modal-import.tsx

```diff
@@ -0,0 +1,115 @@
+import type { ImportBookmarkForm } from "@/entities/bookmark";
+import { Button } from "@/shared/ui/button";
+import { trpc } from "@/shared/trpc";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { FormProvider, useForm } from "react-hook-form";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/shared/ui/dialog";
+import { ImportBookmarkFormUi, importBookmarkSchema } from "@/entities/bookmark";
+import { useParams, useSearchParams } from "react-router-dom";
+
+export function BookmarkModalImport() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const params = useParams();
+  const utils = trpc.useUtils();
+  const { data: collections } = trpc.collections.list.useQuery();
+  const form = useForm<ImportBookmarkForm>({
+    resolver: zodResolver(importBookmarkSchema),
+    defaultValues: {
+      importFile: undefined,
+      collectionId: "",
+    },
+  });
+
+  const create = trpc.bookmarks.import.useMutation({
+    onSuccess() {
+      utils.bookmarks.list.invalidate();
+      utils.collections.list.invalidate();
+      utils.stats.all.invalidate();
+      setSearchParams((prev) => {
+        prev.delete("modal");
+        return prev;
+      });
+      form.reset();
+    },
+  });
+
+  async function submit(data: ImportBookmarkForm) {
+    let importedBookmarks: {url: string, collectionId: string}[] = [];
+
+    // To read the imported bookmark file
+    const reader = new FileReader();
+    
+    reader.onload = function (e) {
+      // To parse the html content of the imported bookmark file
+      const parser = new DOMParser();
+
+      if (e?.target?.result === null || e.target === null){
```

Comment:
This check might be simplified to `if (!e?.target?.result)`. Also instead of return console.log I would notifiy user about that using `toast.warn(...)`.

Reply:
Yes, missed it. Fixed.